src/heaps/binomial_heap.jl

```diff
@@ -0,0 +1,371 @@
+    if b1.data > b2.data
+        b1, b2 = b2, b1
+    end
+    b2.parent = b1 
+    b2.sibling = b1.child 
+    b1.child = b2 
+    b1.degree += 1 
+    return b1
+end
+# adjust! function rearranges the heap so that 
+# heap is in increasing order of degree and 
+# no two binomial trees have same degree in this heap 
+function adjust!(heap::MutableBinomialHeap{T}) where T
+    _heap = heap.rootList
+    n = length(_heap)
+    if n <= 1
+        return
+    end
+    i = 1
+    if n == 2
+        j = 2
+        k = n + 1
+    else
+        j = i + 1
+        k = j + 1
+    end
+    while i <= length(_heap)
+        current_size = length(_heap) + 1
+        if j >= current_size
+            i += 1
+        elseif _heap[i].degree < _heap[j].degree
+            i += 1
+            j += 1
+            if k !== current_size
+                k += 1
+            end
+        elseif (k < current_size)  && (_heap[i].degree == _heap[j].degree) && (_heap[i].degree == _heap[k].degree)
+            i += 1
+            j += 1
+            k += 1  
+        elseif _heap[i].degree == _heap[j].degree
+            temp = merge_binomial_trees!(_heap[i], _heap[j])
+            _heap[i] = temp
+            @inbounds _heap[j] = nothing
+            filter!(e->e ≠ nothing, _heap)
+            if k < (length(_heap) + 1)
+                k = j + 1 
+            end
+        end
+    end
+end
+
+function insert_tree_in_heap!(_heap::MutableBinomialHeap{T}, tree::MutableBinomialHeapNode{T}) where T 
+    temp = MutableBinomialHeap{T}()
+    push!(temp.rootList, tree)
+    _union!(_heap, temp, true)
+    adjust!(_heap)
+    return
+end   
+function remove_min__from_tree!(tree::MutableBinomialHeapNode{T}) where T 
+    heap = Union{Nothing,MutableBinomialHeapNode{T}}[]
+    temp = tree.child; 
+    while temp !== nothing 
+        lo = temp 
+        temp = temp.sibling 
+        lo.sibling = nothing 
+        push!(heap, lo)
+    end 
+    return reverse!(heap) 
+end
+
+function extract_min!(heap::MutableBinomialHeap{T}) where T
+    _heap = heap.rootList
+    new_heap = MutableBinomialHeap{T}()
+    lo = MutableBinomialHeap{T}()
+    temp = top_element_node(heap)
+    it = iterate(_heap)
+    while it !== nothing
+        if it[1] != temp
+            push!(new_heap.rootList, it[1])
+        end
+        it = iterate(_heap, it[2])
+    end
+    lo.rootList = remove_min__from_tree!(temp)
+    new_heap.nodecount = heap.nodecount - 1
+    _union!(new_heap, lo, false)
+    adjust!(new_heap)
+    heap.rootList = new_heap.rootList
+    heap.nodecount -= 1
+    return temp.data
+end
+
+function find_helper(node::Union{MutableBinomialHeapNode{T},Nothing}, i::Int) where T
+    if node === nothing 
+        return nothing
+    end
+    if node.handle == i 
+        return node
+    end
+    res = find_helper(node.child, i)
+    if res !== nothing 
+        return res
+    end
+    return find_helper(node.sibling, i)
```

Comment:
Shorter for this kind of thing to use short-cut evaluation
```suggestion
    node === nothing && return nothing
    node.handle == i && return node
    res = find_helper(node.child, i)
    res  !== nothing && return res
    return find_helper(node.sibling, i)
```